src/librustc_mir/dataflow/generic.rs

```diff
@@ -316,7 +338,7 @@ where
         };
 
         let block_data = &self.body.basic_blocks()[target_block];
-        self.results.analysis.apply_partial_block_effect(
+        self.results.borrow().analysis.apply_partial_block_effect(
```

Comment:
These won't pass the borrow checker since we need to mutably borrow `self.state` and pass it as an argument to `Analysis::apply_partial_block_effect`.